packages/workshop-app/app/root.tsx

```diff
@@ -90,12 +96,16 @@ export default function App() {
 				<ScrollRestoration />
 				<Scripts />
 				<LiveReload />
-				<script dangerouslySetInnerHTML={{ __html: getWebsocketJS() }} />
+				{/*
+				 FIXME: let me know if we need this now that remix have livereload
+				<script dangerouslySetInnerHTML={{ __html: getWebsocketJS() }} /> */}
```

Comment:
We do. This is for handling data changes (the exercise README) that Remix can't know about.

Reply:
when can add all README.mdx (and any file we want) to remix.config.js watchPaths

```js
const workshopRoot = process.env.KCDSHOP_CONTEXT_CWD ?? process.cwd()
...
	watchPaths: [
		'./tailwind.config.ts',
		'./utils/*.*',
		workshopRoot + '/**/README.mdx',
	],
```

that is working (is it slow? 1.2s?)
```shell
[dev:remix] 💿 File changed: ..\example\exercises\01.nested-routing\01.problem.outlet\README.mdx
[dev:remix] 💿 Rebuilding...
[dev:remix] 💿 Rebuilt in 1.2s
[dev:remix] Waiting for app server (ca3ae323)
[dev:remix] App server took 265ms
[dev:remix] > HMR
```
now we need to find a way to trigger the relevant loader from apps.server.ts `handleFileChanges`

Reply:
WebSocketServer is back